modulus/datapipes/climate/climate_hdf5.py

```diff
@@ -0,0 +1,590 @@
+
+            # If channels not provided, use all of them
+            if self.channels is None:
+                self.channels = [i for i in range(f["fields"].shape[1])]
+
+            # If num_samples_per_year use all
+            if self.num_samples_per_year is None:
+                self.num_samples_per_year = data_samples_per_year
+
+            # Adjust image shape if patch_size defined
+            if self.patch_size is not None:
+                self.cropped_data_shape = [
+                    s - s % self.patch_size[i] for i, s in enumerate(self.data_shape)
+                ]
+            else:
+                self.cropped_data_shape = self.data_shape
+            self.logger.info(f"Input data shape: {self.cropped_data_shape}")
+
+            # Get total length
+            self.total_length = self.n_years * self.num_samples_per_year
+
+            # Sanity checks
+            if max(self.channels) >= f["fields"].shape[1]:
+                raise ValueError(
+                    f"Provided channel has indexes greater than the number \
+                of fields {f['fields'].shape[1]}"
+                )
+
+            if self.num_samples_per_year > data_samples_per_year:
+                raise ValueError(
+                    f"num_samples_per_year ({self.num_samples_per_year}) > number of \
+                    samples available ({data_samples_per_year})!"
+                )
+
+            self.logger.info(f"Number of samples/year: {self.num_samples_per_year}")
+            self.logger.info(f"Number of channels available: {f['fields'].shape[1]}")
+
+    def _load_statistics(self) -> None:
+        """Loads climate statistics from pre-computed numpy files
+
+        The statistic files should be of name global_means.npy and global_std.npy with
+        a shape of [1, C, 1, 1] located in the stat_dir.
+
+        Raises
+        ------
+        IOError
+            If mean or std numpy files are not found
+        AssertionError
+            If loaded numpy arrays are not of correct size
+        """
+        # If no stats dir we just skip loading the stats
+        if self.stats_dir is None:
+            self.mu = None
+            self.std = None
+            return
+        # load normalisation values
+        mean_stat_file = self.stats_dir / Path("global_means.npy")
+        std_stat_file = self.stats_dir / Path("global_stds.npy")
+
+        if not mean_stat_file.exists():
+            raise IOError(f"Mean statistics file {mean_stat_file} not found")
+        if not std_stat_file.exists():
+            raise IOError(f"Std statistics file {std_stat_file} not found")
+
+        # has shape [1, C, 1, 1]
+        self.mu = np.load(str(mean_stat_file))[:, self.channels]
+        # has shape [1, C, 1, 1]
+        self.sd = np.load(str(std_stat_file))[:, self.channels]
+
+        if not self.mu.shape == self.sd.shape == (1, len(self.channels), 1, 1):
+            raise AssertionError("Error, normalisation arrays have wrong shape")
```

Comment:
Suggestion for a more informative message:
`raise AssertionError(f"Error, normalisation arrays mean and std have shapes {self.mu.shape}, {self.sd.shape}, but expected arrays of shape {(1, len(self.channels), 1, 1)}")`

Reply:
fixed this